training/abfe.py

```diff
@@ -87,24 +88,29 @@ def convert_uIC50_to_kJ_per_mole(amount_in_uM):
     ff_raw = open(args.forcefield, "r").read()
     ff_handlers = deserialize(ff_raw)
 
-    ports = [
-        50000,
-        50001,
-        50002,
-        50003,
-        50004,
-        50005,
-        # 50006,
-        # 50007,
-        # 50008,
-        # 50009
-    ]
 
+    ip_list = []
+    expected_workers = 6
+    stop_event = threading.Event()
+
+    registry = registration.Registry(ip_list, expected_workers, stop_event)
+
+    server = grpc.server(futures.ThreadPoolExecutor(max_workers=16))
```

Comment:
I’m finding the combination that combination of concurrent and threading odd.  I’m assuming there is a way to use a process pool executor in case you run into GIL issues.

Reply:
This is actually the recommended pattern. The server code is only blocked by IO, not CPU/compute. 

See this thread for a more in-depth discussion:

https://github.com/grpc/grpc/issues/18535